public/app/plugins/datasource/elasticsearch/components/QueryEditor/MetricAggregationsEditor/MetricEditor.test.tsx

```diff
@@ -1,85 +1,112 @@
+        <ElasticsearchProvider
+          datasource={{ getFields } as ElasticDatasource}
+          query={query}
+          range={getDefaultTimeRange()}
+          onChange={() => {}}
+          onRunQuery={() => {}}
+        >
+          {children}
+        </ElasticsearchProvider>
+      );
 
-    render(<MetricEditor value={avg} />, { wrapper });
+      render(<MetricEditor value={avg} />, { wrapper });
 
-    act(() => {
-      fireEvent.click(screen.getByText('Select Field'));
+      act(() => {
+        fireEvent.click(screen.getByText('Select Field'));
+      });
+
+      expect(await screen.findByText('None')).toBeInTheDocument();
     });
 
-    expect(await screen.findByText('None')).toBeInTheDocument();
-  });
+    it('Should not display a "None" option for "field" if the metric does not support inline script', async () => {
+      const avg: UniqueCount = {
+        id: '1',
+        type: 'cardinality',
+      };
 
-  it('Should not display a "None" option for "field" if the metric does not support inline script', async () => {
-    const avg: UniqueCount = {
-      id: '1',
-      type: 'cardinality',
-    };
+      const query: ElasticsearchQuery = {
+        refId: 'A',
+        query: '',
+        metrics: [avg],
+        bucketAggs: [defaultBucketAgg('2')],
+      };
 
-    const query: ElasticsearchQuery = {
-      refId: 'A',
-      query: '',
-      metrics: [avg],
-      bucketAggs: [defaultBucketAgg('2')],
-    };
+      const getFields: ElasticDatasource['getFields'] = jest.fn(() => from([[]]));
 
-    const getFields: ElasticDatasource['getFields'] = jest.fn(() => from([[]]));
+      const wrapper = ({ children }: { children: ReactNode }) => (
+        <ElasticsearchProvider
+          datasource={{ getFields } as ElasticDatasource}
+          query={query}
+          range={getDefaultTimeRange()}
+          onChange={() => {}}
+          onRunQuery={() => {}}
+        >
+          {children}
+        </ElasticsearchProvider>
+      );
 
-    const wrapper = ({ children }: { children: ReactNode }) => (
-      <ElasticsearchProvider
-        datasource={{ getFields } as ElasticDatasource}
-        query={query}
-        range={getDefaultTimeRange()}
-        onChange={() => {}}
-        onRunQuery={() => {}}
-      >
-        {children}
-      </ElasticsearchProvider>
-    );
+      render(<MetricEditor value={avg} />, { wrapper });
 
-    render(<MetricEditor value={avg} />, { wrapper });
+      act(() => {
+        fireEvent.click(screen.getByText('Select Field'));
+      });
 
-    act(() => {
-      fireEvent.click(screen.getByText('Select Field'));
+      expect(await screen.findByText('No options found')).toBeInTheDocument();
+      expect(screen.queryByText('None')).not.toBeInTheDocument();
     });
+  });
 
-    expect(await screen.findByText('No options found')).toBeInTheDocument();
-    expect(screen.queryByText('None')).not.toBeInTheDocument();
+  describe('getTypeOptions()', () => {
```

Comment:
I usually try to avoid exporting things just for testing, for example in this very same file I'm testing for the field picker to be nullable in certain conditions by mounting the whole MetricEditor.

IMHO it's a more reliable way to test as it actually tests the component the same way a user would use it. what do you think?

Reply:
Works for me